### ISSUE TITLE
代码

### ISSUE THREAD

**Issue by she-xueying** [iCVTEAM/LRT#1]
I can't reproduce the results in the article. I'd like to ask how you set the parameters.

**Comment by zhao1f** [comment]
Feel free to send emails with training logs : )

**Comment by zhao1f** [comment]
Sorry... it seems I put the command of the incremental stage before the base learning stage in the readme File.

**Comment by she-xueying** [comment]
> Sorry... it seems I put the command of the incremental stage before the base learning stage in the readme File.

I first carried out the instructions for the basic learning of the advanced nature, and then carried out the instructions for the incremental stage. However, I still can't achieve the results described in the article. How can I solve this problem?

**Comment by zhao1f** [comment]
Send emails to zhaoyf@buaa.edu.cn with details

**Comment by zhao1f** [comment]
> > Sorry... it seems I put the command of the incremental stage before the base learning stage in the readme File.
> 
> I first carried out the instructions for the basic learning of the advanced nature, and then carried out the instructions for the incremental stage. However, I still can't achieve the results described in the article. How can I solve this problem?

I really want to help you, but the information provided is too little, so it is difficult to solve the problem.
I carefully checked my running instructions, and I guess the reason may be as follows:

In order to run quickly, I set the training base epoch of the base training process to 0, which means that the model is not trained at all.

Therefore, I modified the readme file, please set the base training epoch, for example, 100 for miniimagenet, 60~70 for ImageNet100.

**Comment by she-xueying** [comment]
I'm sorry, I didn't make myself clear.
I didn't use the parameter settings in the README. For both the base class stage and the incremental stage, I used the default parameters set in the `train` method. At the same time, I added the `-moder_dir` parameter during the incremental stage, but I still couldn't achieve the same results as those in the article. 
The results I replicated differ from the results in the article by about 10%. What could be the cause of this issue, and how can I solve it? 
I would appreciate it if you could give me some guidance. Thank you so much!  




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "iCVTEAM/LRT"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年4月18日(星期五) 下午2:19
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [iCVTEAM/LRT] 代码 (Issue #1)



     
Sorry... it seems I put the command of the incremental stage before the base learning stage in the readme File.
  
I first carried out the instructions for the basic learning of the advanced nature, and then carried out the instructions for the incremental stage. However, I still can't achieve the results described in the article. How can I solve this problem?
  
I really want to help you, but the information provided is too little, so it is difficult to solve the problem.
 I carefully checked my running instructions, and I guess the reason may be as follows:
 
In order to run quickly, I set the training base epoch of the base training process to 0, which means that the model is not trained at all.
 
Therefore, I modified the readme file, please set the base training epoch, for example, 100 for miniimagenet, 60~70 for ImageNet100.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;
  zhao1f left a comment (iCVTEAM/LRT#1)
   
Sorry... it seems I put the command of the incremental stage before the base learning stage in the readme File.
  
I first carried out the instructions for the basic learning of the advanced nature, and then carried out the instructions for the incremental stage. However, I still can't achieve the results described in the article. How can I solve this problem?
  
I really want to help you, but the information provided is too little, so it is difficult to solve the problem.
 I carefully checked my running instructions, and I guess the reason may be as follows:
 
In order to run quickly, I set the training base epoch of the base training process to 0, which means that the model is not trained at all.
 
Therefore, I modified the readme file, please set the base training epoch, for example, 100 for miniimagenet, 60~70 for ImageNet100.

**Comment by zhao1f** [comment]
Send emails to zhaoyf@buaa.edu.cn with detailed logs.

Using train.py will lead to several hyperparameters being inconsistent.  At least the following hyper-params must be refined for miniimagenet.


-base_mode 'ft_cos' 
-new_mode 'ft_cos'   (default='avg_cos', which means the final transferring layers are not trained) [important]
-lr_base 0.01 (the default setting is 0.1, which is too large) [important]
 -lr_new 0.01 (0.001 might be optional)  [important]
 -epochs_base 100 
-epochs_new 20 [default is 100, which is too large]
 -milestones 40 70 
 -gpu 5,7 [default='0,1,2,3',  please use 2 gpus, the batchsize and gpu numbers will affect the final performance]

**Comment by she-xueying** [comment]
Thank you for your explanation. I will give it a try.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "iCVTEAM/LRT"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年4月21日(星期一) 下午3:28
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [iCVTEAM/LRT] 代码 (Issue #1)



   
Send emails to ***@***.*** with detailed logs.
 
Using train.py will lead to several hyperparameters being inconsistent.  At least the following hyper-params must be refined for miniimagenet.
 
-base_mode 'ft_cos'
 -new_mode 'ft_cos'   (default='avg_cos', which means the final transferring layers are not trained) [important]
 -lr_base 0.01 (the default setting is 0.1, which is too large) [important]
 -lr_new 0.01 (0.001 might be optional)  [important]
 -epochs_base 100
 -epochs_new 20 [default is 100, which is too large]
 -milestones 40 70
 -gpu 5,7 [default='0,1,2,3',  please use 2 gpus, the batchsize and gpu numbers will affect the final performance]

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;
  zhao1f left a comment (iCVTEAM/LRT#1)
 
Send emails to ***@***.*** with detailed logs.
 
Using train.py will lead to several hyperparameters being inconsistent.  At least the following hyper-params must be refined for miniimagenet.
 
-base_mode 'ft_cos'
 -new_mode 'ft_cos'   (default='avg_cos', which means the final transferring layers are not trained) [important]
 -lr_base 0.01 (the default setting is 0.1, which is too large) [important]
 -lr_new 0.01 (0.001 might be optional)  [important]
 -epochs_base 100
 -epochs_new 20 [default is 100, which is too large]
 -milestones 40 70
 -gpu 5,7 [default='0,1,2,3',  please use 2 gpus, the batchsize and gpu numbers will affect the final performance]

**Comment by she-xueying** [comment]
I'm sorry to trouble you again. After using the parameters you provided, the accuracy of the model still can't reach the effect described in the article. 
I found that the coefficient of the loss_clip loss in the total loss is 0. Could this be related to the problem? 
After deleting this coefficient, I trained the model again, but it still can't reach the accuracy stated in the article. What could be the cause of this?
I hope you can give me some guidance. Thank you for your help!  




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "iCVTEAM/LRT"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年4月21日(星期一) 下午3:28
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [iCVTEAM/LRT] 代码 (Issue #1)



   
Send emails to ***@***.*** with detailed logs.
 
Using train.py will lead to several hyperparameters being inconsistent.  At least the following hyper-params must be refined for miniimagenet.
 
-base_mode 'ft_cos'
 -new_mode 'ft_cos'   (default='avg_cos', which means the final transferring layers are not trained) [important]
 -lr_base 0.01 (the default setting is 0.1, which is too large) [important]
 -lr_new 0.01 (0.001 might be optional)  [important]
 -epochs_base 100
 -epochs_new 20 [default is 100, which is too large]
 -milestones 40 70
 -gpu 5,7 [default='0,1,2,3',  please use 2 gpus, the batchsize and gpu numbers will affect the final performance]

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;
  zhao1f left a comment (iCVTEAM/LRT#1)
 
Send emails to ***@***.*** with detailed logs.
 
Using train.py will lead to several hyperparameters being inconsistent.  At least the following hyper-params must be refined for miniimagenet.
 
-base_mode 'ft_cos'
 -new_mode 'ft_cos'   (default='avg_cos', which means the final transferring layers are not trained) [important]
 -lr_base 0.01 (the default setting is 0.1, which is too large) [important]
 -lr_new 0.01 (0.001 might be optional)  [important]
 -epochs_base 100
 -epochs_new 20 [default is 100, which is too large]
 -milestones 40 70
 -gpu 5,7 [default='0,1,2,3',  please use 2 gpus, the batchsize and gpu numbers will affect the final performance]

**Comment by zhao1f** [comment]
Send emails to [zhaoyf@buaa.edu.cn](mailto:zhaoyf@buaa.edu.cn) with detailed logs.   I cannot locate the problems with so little information.
